polygeist/tools/cgeist/Test/Verification/sycl/builtin_types.cpp

```diff
@@ -0,0 +1,57 @@
+// RUN: clang++ -fsycl -fsycl-device-only -O0 -w -emit-mlir -o - %s | FileCheck %s
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+// CHECK-LABEL: func.func @_Z13opencl_float2Dv2_f(
+// CHECK:         %arg0: memref<?x2xf32> {llvm.noundef})
+SYCL_EXTERNAL void opencl_float2(__cl_float2 var) {}
+
+// CHECK-LABEL: func.func @_Z13opencl_float4Dv4_f(
+// CHECK:         %arg0: memref<?x4xf32> {llvm.noundef})
+SYCL_EXTERNAL void opencl_float4(__cl_float4 var) {}
```

Comment:
These should be vectors (`vector<2xf32>` and `vector<4xf32>`)

Reply:
done

Reply:
@victor-eds Do you mean the codegen of `__cl_float2` should be `vector<2xf32>` instead of `memref<?x2xf32>`?

Reply:
fixed in latest commit.